### PR TITLE
Add mobile child attendance reservation edit

### PR DESCRIPTION
**Pull request by patari**

### CODE REVIEW

File: frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+    form.state.times.every(
+      (reservation) =>
+        reservation.startTime !== '' && reservation.endTime !== ''
+    )
+      ? boundTimes.length === 1
+        ? 'same-row'
+        : boundTimes.length > 1
+        ? 'next-row'
+        : undefined
+      : undefined
```
Comment (akheron) [inline frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx:432]: The maximum amount of reservations per day is 2. This is not enforced by constraints or such, but I think it is assumed somewhere in the backend logic. A citizen cannot add more than 2 reservations anyway.

File: frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+
+  return (
+    <div data-qa="reservation-text">
+      {reservation.absenceType !== null ? (
+        <ServiceTime>{i18n.attendances.types.ABSENT}</ServiceTime>
+      ) : (
+        <AttendanceDailyServiceTimes
+          hideLabel={true}
+          times={reservation.dailyServiceTimes}
+          reservations={reservation.reservations}
```
Comment (akheron) [inline frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx:338]: Reservations are shown in the wrong order now if the user enters them in the wrong order.

**Review by akheron — COMMENTED** [review]
The UI is pretty broken on a 360 px wide device now.

Screenshot 2023-11-17 at 15 03 16

Also, it took some time to understand the concept of "non-reservable reservation". Would "reservation in non-reservable range" or something be too long?

File: frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx (hunk 2 above)
Comment (patari) [reply frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx]: Fixed

**Comment by patari** [comment]
> The UI is pretty broken on a 360 px wide device now.

Fixed

> Also, it took some time to understand the concept of "non-reservable reservation". Would "reservation in non-reservable range" or something be too long?

Yeah naming is tough on this one. I don't think reservation should be used here at all because it also contains absences and daily service times. First I was thinking like `DayPlan` or something but I don't know yet. Let's check next week!

**Review by akheron — APPROVED** [review]

**Pull request merged**